cmake/dds_idl_sources.cmake

```diff
@@ -136,7 +139,8 @@ function(_opendds_target_idl_sources target)
     get_property(generated_dependencies SOURCE ${idl_file}
       PROPERTY OPENDDS_IDL_GENERATED_DEPENDENCIES SET)
 
-    if(generated_dependencies)
+    # TODO Fix this
```

Comment:
I think this has to be left for later. Enabling this again breaks the OpenDDS build because of [these two `opendds_target_sources` calls in the Rtps CMakeLists](https://github.com/iguessthislldo/OpenDDS/blob/43f0679b10cb66981cfe74125251bee76ac42f3d/dds/DCPS/RTPS/CMakeLists.txt#L52-L69). This should be valid, but if this check wasn't disabled then it only runs the `tao_idl`-only. I think this is an attempt to keep the IDL compilers from running more than once in cases like here:

https://github.com/OpenDDS/OpenDDS/blob/eb011d68c99cf6096465bb34f3539eda9a5a6ac7/tests/cmake/Messenger/Messenger_1/CMakeLists.txt#L39-L50

However the caller could pass different options like in RTPS and that should be valid.  This doesn't completely fix that issue though if there are conflicting options on two different targets. That would require per-target isolated generation which the `opendds_target_syntax` implies, but doesn't do.

I will update the comment with this information.

Reply:
#4306